src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,146 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.exceptions.RegistrationException;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static final String VALID_LOGIN = "validLogin";
+    private static final String VALID_PASSWORD = "validPassword";
+    private static final String EXISTING_USER_LOGIN = "existingUserLogin";
+    private static final String SHORT_LOGIN = "log";
+    private static final String SHORT_PASSWORD = "pass";
+    private static final String EMPTY_STRING = "";
+    private static final int VALID_AGE = 24;
+    private static final int INVALID_AGE = 15;
+
+    private RegistrationService registrationValidator;
+    private StorageDao storageDao;
```

Comment:
Since the `RegistrationServiceImpl` and `StorageDaoImpl` instances are not dependent on any instance state and do not seem to change between tests, you should consider using `@BeforeAll` with static initialization instead of `@BeforeEach`. This would improve the performance of your tests as they would not need to create a new instance before each test.